subt/main.py

```diff
@@ -416,7 +416,13 @@ def return_home(self, timeout, home_threshold=None):
                     desired_direction += math.pi  # symmetry
                     for angle in self.joint_angle_rad:
                         desired_direction -= angle
-                self.bus.publish('desired_altitude', target_z)
+
+                d = distance3D(self.xyz, [target_x, target_y, target_z])
+                time_to_target = d/self.max_speed
```

Comment:
can you be in target, i.e. division by zero on the following line?

Reply:
in theory, yes, but it should not happen since we pick the targets from the trace and when there is nothing to pick from we jump out of the while loop - do you think it is worth checking it here as well?

Reply:
can you rebase? I see obsolete code above
```
while distance3D(self.xyz, (0, 0, 0)) > HOME_THRESHOLD and self.sim_time_sec - start_time < timeout.total_seconds():
```

Reply:
done